qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -89,13 +93,217 @@ def compute_probabilities(self, circuits: List[QuantumCircuit])
+                "phase": exp_helper.iq_phase(experiment_circuits),
+                "num_circuits": len(experiment_circuits),
+            }
+
+        return prob_help_list
+
+    def _verify_parameters(self):
+        """Check parameters before computing probability"""
+        if self.exp_helper_list is None:
+            raise ValueError("Please set the experiment helper list.")
+        if self.exp_list is None:
+            raise ValueError("Please set the experiment list.")
+
+        number_of_experiments = len(self.exp_list)
+        number_of_helpers = len(self.exp_helper_list)
+
+        if number_of_experiments == 0:
+            raise ValueError("The experiment list cannot be empty.")
+        if number_of_helpers == 0:
+            raise ValueError("The experiment helper list cannot be empty.")
+
+        if number_of_experiments != number_of_helpers:
+            raise ValueError(
+                "The number of helpers {} and the number of experiment {} don't match.".format(
+                    number_of_experiments, number_of_helpers
+                )
+            )
+
+    def _parallel_exp_circ_splitter(self, qc_list: List[QuantumCircuit]):
+        """
+        Splits a quantum circuits to its parallel components.
+        Args:
+            qc_list: The list of quantum circuit the backend get as input.
+
+        Returns:
+            List: A List for each experiment. each entry is a list of quantum circuits relevant to
+            the same experiment.
+
+        Raises:
+            QiskitError: If an instruction is applied with qubits that aren't under the same experiment.
+        """
+        # exp_idx_map is to connect between experiment and its circuit in the output.
+        exp_idx_map = {exp: exp_idx for exp_idx, exp in enumerate(self.exp_list)}
+        qubit_exp_map = self._create_qubit_exp_map()
+
+        exp_circuits_list = [[] for _ in self.exp_list]
+
+        for qc in qc_list:
+            # Quantum Register to qubit mapping
+            qubit_indices = {bit: idx for idx, bit in enumerate(qc.qubits)}
+
+            # initialize quantum circuit for each experiment for this instance of circuit to fill
+            # with instructions.
+            for exp_circuit in exp_circuits_list:
+                qcirc = qc.copy()
+                qcirc.data.clear()
+                qcirc.metadata.clear()
+                exp_circuit.append(qcirc)
+
+            # fixing metadata
+            for exp_metadata in qc.metadata["composite_metadata"]:
+                # getting a qubit of one of the experiment that we ran in parallel
+                qubit = qubit_exp_map[exp_metadata["qubits"][0]]
+                # using the qubit to access the experiment. Then, we go to the last circuit in
+                # `exp_circuit` of the corresponding experiment, and we overwrite the metadata.
+                exp_circuits_list[exp_idx_map[qubit]][-1].metadata = exp_metadata.copy()
```

Comment:
My bad. I am changing `qubit` to `exp`:
```python
# getting a qubit of one of the experiment that we ran in parallel
exp = qubit_exp_map[exp_metadata["qubits"][0]]
# using the qubit to access the experiment. Then, we go to the last circuit in
# `exp_circuit` of the corresponding experiment, and we overwrite the metadata.
exp_circuits_list[exp_idx_map[exp]][-1].metadata = exp_metadata.copy()
```

The reason for the last circuit is because the last circuit in the list of each experiment corresponds to `qc` (which is the circuit in the loop)